src/DataGrid.php

```diff
@@ -384,15 +384,24 @@ public function render()
 		}
 
 		$callback = $this->rowCallback ?: NULL;
+		$hasGroupActionOnRows = FALSE;
 
 		foreach ($items as $item) {
 			$rows[] = $row = new Row($this, $item, $this->getPrimaryKey());
 
+			if(!$hasGroupActionOnRows && $row->hasGroupAction()){
+				$hasGroupActionOnRows = TRUE;
+			}
+			
 			if ($callback) {
 				$callback($item, $row->getControl());
 			}
 		}
 
+		if($hasGroupActionOnRows){
+			$hasGroupActionOnRows = $this->hasGroupActions();
```

Comment:
The problem is that `$row->hasGroupAction()` returns true if no group actions are defined, so this checks at the end if group actions are actually there since $hasGroupActionOnRows will be true and  grid maybe does not have group actions at all. If we skip this then it's possible to have screwed up grid (checkboxes on rows, and no group actions at all)

Solution was either that or to go with new loop if group actions are available, thought that it's going to be faster with one loop only, although many grids does not contain group actions at all.
if you want I can refactor so that if group actions exists we do a new loop and set it there, probably more readable option....